source/hp/dof_handler.cc

```diff
@@ -1269,6 +1276,40 @@ namespace hp
       active_fe_indices[i] = cell->active_fe_index();
   }
 
+  template <int dim, int spacedim>
+  void DoFHandler<dim,spacedim>::initialize(const Triangulation<dim,spacedim> &tria,
+                                            const hp::FECollection<dim,spacedim> &fe)
+  {
+    this->tria = &tria;
+
+    if (dynamic_cast<const parallel::shared::Triangulation< dim, spacedim>*> (&*this->tria) != nullptr)
```

Comment:
Most of this is copied from the other constructor -- could you break this common code out into a separate function to minimize the code duplication?